frontend/src/routes/edit/musics/+page.svelte

```diff
@@ -0,0 +1,151 @@
+<!--
+SPDX-FileCopyrightText: 2023 Marlon W (Mawoka)
+
+SPDX-License-Identifier: MPL-2.0
+-->
+
+<script lang="ts">
+	import BrownButton from '$lib/components/buttons/brown.svelte';
+	import { navbarVisible } from '$lib/stores';
+	import { getLocalization } from '$lib/i18n';
+	const { t } = getLocalization();
+
+	let file_input: HTMLInputElement;
+	navbarVisible.set(false);
+
+	let stats: { progress: number; time_elapsed: number; speed: number } = {
+		progress: 0,
+		time_elapsed: 0,
+		speed: 0
+	};
+
+	let upload_stats: { progress: number; upload_started: undefined | Date; time_elapsed: number } =
+		{
+			progress: 0,
+			upload_started: undefined,
+			time_elapsed: 0
+		};
+
+	const speed_extraction_regex = /speed=(\d+\.\d+)x(?![\s\S]*speed=\d+\.\d+x)/;
+	let last_time_elapsed = 0;
+
+	// eslint-disable-next-line no-unused-vars
+	enum Status {
+		// eslint-disable-next-line no-unused-vars
+		Idle,
+		// eslint-disable-next-line no-unused-vars
+		Compressing,
+		// eslint-disable-next-line no-unused-vars
+		CompressDone,
+		// eslint-disable-next-line no-unused-vars
+		Uploading,
+		// eslint-disable-next-line no-unused-vars
+		Done
+	}
+
+	let status: Status = Status.Idle;
+
+	let file_size_in_mi: undefined | number = undefined;
+	let original_file_size_in_mi: undefined | number = undefined;
+	let file: File;
+	let file_data: undefined | Blob = undefined;
+
+	const compress_music = async () => {
+		if (status !== Status.Idle) {
+			return;
+		}
+		status = Status.Compressing;
+		file = file_input.files[0];
+		original_file_size_in_mi = file.size / 1_048_576;
+		file_size_in_mi = file.size / 1_048_576;
+		file_data = file;
+		status = Status.CompressDone;
+		stats.progress = 1;
+	};
+
+	const upload_music = async () => {
+		status = Status.Uploading;
+		upload_stats.upload_started = new Date();
+		const xhr = new XMLHttpRequest();
+		const success = await new Promise((resolve) => {
+			xhr.upload.addEventListener('progress', (event) => {
+				if (event.lengthComputable) {
+					upload_stats.progress = event.loaded / event.total;
+					upload_stats.time_elapsed = (new Date() - upload_stats.upload_started) / 1000;
+				}
+			});
+			xhr.addEventListener('loadend', () => {
+				resolve({
+					success: xhr.readyState === 4 && xhr.status === 200,
+					body: xhr.responseText
+				});
+			});
+			xhr.open('POST', `/api/v1/storage/raw/${file.name}`, true);
+			xhr.setRequestHeader('Content-Type', 'music/mp3');
```

Comment:
Not always mp3

Reply:
You're right but I don't have any other option (checking file extension?)